packages/astro/src/compiler/codegen/index.ts

```diff
@@ -55,11 +55,13 @@ interface HydrationAttributes {
 function findHydrationAttributes(attrs: Record<string, string>): HydrationAttributes {
   let method: HydrationAttributes['method'];
 
-  const hydrationDirectives = new Set([':load', ':idle', ':visible'])
+  const hydrationRegex = /(?<=client:)[^$]*/;
+  const hydrationDirectives = new Set(['load', 'idle', 'visible']);
 
   for (const [key, val] of Object.entries(attrs)) {
-    if (!key.startsWith(':')) continue;
-    if (hydrationDirectives.has(key) && val === 'true') method= key.slice(1) as HydrationAttributes['method'];
+    const matches = key.match(hydrationRegex);
```

Comment:
I know this method is more DRY and that's nice but parsing the string with a regex is extra work we don't really need, we can just directly check the key by adding `client:` to each of the methods in the set:

```js
const hydrationDirectives = new Set(['client:load', 'client:idle', 'client:visible']);
```

Then do the slice only if there's a match.

Reply:
Yea that's a good point, I was getting a little too fancy there leaning on the regex to match since that would run against every single attribute that goes through

Reply:
Moved back to simple string compare with b290972ae350d072532ff13365ccc3a002004522